pyglmnet/distributions.py

```diff
@@ -0,0 +1,435 @@
+        theta = self.theta
+        log_likelihood = \
+            np.sum(loggamma(y + theta) - loggamma(theta) - loggamma(y + 1) +
+                   theta * np.log(theta) + y * np.log(y_hat) - (theta + y) *
+                   np.log(y_hat + theta))
+        return log_likelihood
+
+    def grad_log_likelihood(self, X, y, beta0, beta):
+        """Gradient of L2-penalized log likelihood."""
+        z = self._z(beta0, beta, X)
+        mu = self.mu(z)
+        grad_mu = self.grad_mu(z)
+        theta = self.theta
+        partial_beta_0 = grad_mu * ((theta + y) / (mu + theta) - y / mu)
+        grad_beta0 = np.sum(partial_beta_0)
+        grad_beta = np.dot(partial_beta_0.T, X)
+        return grad_beta0, grad_beta
+
+    def gradhess_log_likelihood_1d(self, xk, y, beta0, beta):
+        """One-dimensional Gradient and Hessian of log likelihood."""
+        z = self._z(beta0, beta, X)
+        mu = self.mu(z)
+        grad_mu = expit(z)
+        hess_mu = np.exp(-z) * expit(z) ** 2
+
+        gradient_beta_j = -grad_mu * (y / mu - (y + theta) / (mu + theta))
+        partial_beta_0_1 = hess_mu * (y / mu - (y + theta) / (mu + theta))
+        partial_beta_0_2 = grad_mu ** 2 * \
+            ((y + theta) / (mu + theta) ** 2 - y / mu ** 2)
+        partial_beta_0 = -(partial_beta_0_1 + partial_beta_0_2)
+        gk = np.dot(gradient_beta_j.T, xk)
+        hk = np.dot(partial_beta_0.T, xk ** 2)
+        return gk, hk
+
+
+class Binomial(BaseDistribution):
+    """Class for binomial distribution."""
+
+    def __init__(self):
+        """init."""
+        pass
+
+    def mu(self, z):
+        """Inverse link function."""
+        mu = expit(z)
+        return mu
+
+    def grad_mu(self, z):
+        """Gradient of inverse link."""
+        grad_mu = expit(z) * (1 - expit(z))
+        return grad_mu
+
+    def log_likelihood(self, y, y_hat, z=None):
+        """Log L2-penalized likelihood."""
+        # prevents underflow
+        if z is not None:
+            log_likelihood = np.sum(y * z - np.log(1 + np.exp(z)))
+        # for scoring
+        else:
+            log_likelihood = \
+                np.sum(y * np.log(y_hat) + (1 - y) * np.log(1 - y_hat))
+        return log_likelihood
+
+    def grad_log_likelihood(self, X, y, beta0, beta):
+        """Gradient of L2-penalized log likelihood."""
+        z = self._z(beta0, beta, X)
+        mu = self.mu(z)
+        grad_beta0 = np.sum(mu - y)
+        grad_beta = np.dot((mu - y).T, X).T
+        return grad_beta0, grad_beta
+
+    def gradhess_log_likelihood_1d(self, xk, y, beta0, beta):
+        """One-dimensional Gradient and Hessian of log likelihood."""
+        z = self._z(beta0, beta, X)
+        mu = self.mu(z)
+        gk = np.sum((mu - y) * xk)
+        hk = np.sum(mu * (1.0 - mu) * xk * xk)
+        return gk, hk
+
+
+class Probit(BaseDistribution):
+    """Class for probit distribution."""
+
+    def __init__(self):
+        """init."""
+        pass
```

Comment:
you probably don't need to redefine it since you're inheriting from `BaseDistribution` ?